connectors/src/connectors/slack/temporal/workflows.ts

```diff
@@ -180,26 +185,37 @@ export async function syncOneMessageDebounced(
 export async function memberJoinedChannel(
   connectorId: string,
   nangoConnectionId: string,
-  dataSourceConfig: DataSourceConfig,
-  channelId: string
+  dataSourceConfig: DataSourceConfig
 ): Promise<void> {
-  const slackAccessToken = await getAccessToken(nangoConnectionId);
-  const channel = await getChannel(slackAccessToken, channelId);
-  if (!channel.name) {
-    throw new Error(`Could not find channel name for channel ${channelId}`);
-  }
-  const channelName = channel.name;
-  await executeChild(syncOneChannel.name, {
-    workflowId: syncOneChanneWorkflowlId(connectorId, channelId),
-    args: [
-      connectorId,
-      nangoConnectionId,
-      dataSourceConfig,
-      channelId,
-      channelName,
-    ],
-  });
+  const channelsToJoin: string[] = [];
+  setHandler(
+    botJoinedChanelSignal,
+    ({ channelId }: botJoinedChanelSignalInput) => {
+      if (channelsToJoin.indexOf(channelId) === -1) {
+        channelsToJoin.push(channelId);
+      }
+    }
+  );
 
+  let channelId: string | undefined;
+  while ((channelId = channelsToJoin.shift())) {
+    const slackAccessToken = await getAccessToken(nangoConnectionId);
+    const channel = await getChannel(slackAccessToken, channelId);
+    if (!channel.name) {
+      throw new Error(`Could not find channel name for channel ${channelId}`);
+    }
+    const channelName = channel.name;
+    await executeChild(syncOneChannel.name, {
+      workflowId: syncOneChanneWorkflowlId(connectorId, channelId),
+      args: [
+        connectorId,
+        nangoConnectionId,
+        dataSourceConfig,
+        channelId,
+        channelName,
+      ],
+    });
+  }
   await saveSuccessSyncActivity(connectorId);
```

Comment:
as per your previous comments in the other workflow aren't we supposed to have 0 await after the end of the loop?

Reply:
I just removed the line. It is called in syncOneChannel() anyways.
I need to double check something before adjusting all the comments. I am actually not 100% sure of what I said last time about the handling of these signals. I will explain a bit later.